app/exec/build-tasks-upload.ts

```diff
@@ -25,33 +27,55 @@ export class BuildTaskUpload extends cmdm.TfCommand {
     public exec(args: string[], options: cm.IOptions): any {
         var taskPath = args[0] || options['taskPath'];
         this.checkRequiredParameter(taskPath, 'taskPath', 'path to the task folder');
+        var deferred = Q.defer<agentifm.TaskDefinition>();
+        fs.readFile(path.join(taskPath, 'task.json'), 'utf8', (err, data) => {
+            if(err) {
+                return console.log(err);
+            }
 
-        var result: any = {};
-        result.task = { name: name };
-        var archive = archiver('zip');
-        archive.directory(taskPath, false);
-        //result.source = taskPath;
+            var taskJson = JSON.parse(data);
 
-        var deferred = Q.defer<agentifm.TaskDefinition>();
-        var agentapi = this.getWebApi().getTaskAgentApi(this.connection.accountUrl);
+            var vn = (taskJson.name  || taskPath);
 
-        agentapi.uploadTaskDefinition(archive, null, "", name, true, (err, statusCode, task) => {
-            if(err) {
-                err.statusCode = statusCode;
-                deferred.reject(err);
+            if (!taskJson.id || !check.isUUID(taskJson.id)) {
+                return console.log(vn + ': id is a required guid');
+            }
+
+            if (!taskJson.name || !check.isAlphanumeric(taskJson.name)) {
+                return console.log(vn + ': name is a required alphanumeric string');
             }
-            else {
-                deferred.resolve(<agentifm.TaskDefinition>{
-                    name: name,
-                    sourceLocation: taskPath
-                })
+
+            if (!taskJson.friendlyName || !check.isLength(taskJson.friendlyName, 1, 40)) {
+                return console.log(vn + ': friendlyName is a required string <= 40 chars');
+            }
+
+            if (!taskJson.instanceNameFormat) {
+                return console.log(vn + ': instanceNameFormat is required');    
             }
+
+            var archive = archiver('zip');
+            archive.directory(taskPath, false);
```

Comment:
What happens in error conditions here.  Does it throw?  Do you check an error code?